

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>besca &mdash; BESCA 2.2 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/graphviz.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/custom.css" type="text/css" />
  <link rel="stylesheet" href="_static/gallery.css" type="text/css" />
  <link rel="stylesheet" href="_static/gallery-binder.css" type="text/css" />
  <link rel="stylesheet" href="_static/gallery-dataframe.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Welcome to bescaâ€™s documentation!" href="index.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> BESCA
          

          
          </a>

          
            
            
              <div class="version">
                2.2
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">besca</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#helper-functions">helper functions</a><ul class="simple">
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#preprocessing">preprocessing</a><ul>
<li class="toctree-l3"><a class="reference internal" href="preprocessing/besca.pp.filter.html">filter</a></li>
<li class="toctree-l3"><a class="reference internal" href="preprocessing/besca.pp.filter_gene_list.html">filter_gene_list</a></li>
<li class="toctree-l3"><a class="reference internal" href="preprocessing/besca.pp.frac_pos.html">frac_pos</a></li>
<li class="toctree-l3"><a class="reference internal" href="preprocessing/besca.pp.frac_reads.html">frac_reads</a></li>
<li class="toctree-l3"><a class="reference internal" href="preprocessing/besca.pp.mean_expr.html">mean_expr</a></li>
<li class="toctree-l3"><a class="reference internal" href="preprocessing/besca.pp.top_expressed_genes.html">top_expressed_genes</a></li>
<li class="toctree-l3"><a class="reference internal" href="preprocessing/besca.pp.fraction_counts.html">fraction_counts</a></li>
<li class="toctree-l3"><a class="reference internal" href="preprocessing/besca.pp.top_counts_genes.html">top_counts_genes</a></li>
<li class="toctree-l3"><a class="reference internal" href="preprocessing/besca.pp.normalize_geometric.html">normalize_geometric</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#plotting">plotting</a><ul>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.kp_genes.html">kp_genes</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.kp_counts.html">kp_counts</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.kp_cells.html">kp_cells</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.max_counts.html">max_counts</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.max_genes.html">max_genes</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.max_mito.html">max_mito</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.dropouts.html">dropouts</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.detected_genes.html">detected_genes</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.library_size.html">library_size</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.librarysize_overview.html">librarysize_overview</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.transcript_capture_efficiency.html">transcript_capture_efficiency</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.top_genes_counts.html">top_genes_counts</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.gene_expr_split.html">gene_expr_split</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.gene_expr_split_stacked.html">gene_expr_split_stacked</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.stacked_split_violin.html">stacked_split_violin</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.celllabel_quant_boxplot.html">celllabel_quant_boxplot</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.celllabel_quant_stackedbar.html">celllabel_quant_stackedbar</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.dot_heatmap.html">dot_heatmap</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.dot_heatmap_split.html">dot_heatmap_split</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.dot_heatmap_split_greyscale.html">dot_heatmap_split_greyscale</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.update_qualitative_palette.html">update_qualitative_palette</a></li>
<li class="toctree-l3"><a class="reference internal" href="plotting/besca.pl.nomenclature_network.html">nomenclature_network</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#tools">tools</a><ul>
<li class="toctree-l3"><a class="reference internal" href="tools/besca.tl.count_occurrence.html">count_occurrence</a></li>
<li class="toctree-l3"><a class="reference internal" href="tools/besca.tl.count_occurrence_subset.html">count_occurrence_subset</a></li>
<li class="toctree-l3"><a class="reference internal" href="tools/besca.tl.count_occurrence_subset_conditions.html">count_occurrence_subset_conditions</a></li>
<li class="toctree-l3"><a class="reference internal" href="tools/besca.tl.annotate_cells_clustering.html">annotate_cells_clustering</a></li>
<li class="toctree-l3"><a class="reference internal" href="#toolkits">toolkits</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#batch-correction">batch correction</a><ul>
<li class="toctree-l5"><a class="reference internal" href="bcor/besca.tl.bcor.batch_correct.html">batch_correct</a></li>
<li class="toctree-l5"><a class="reference internal" href="bcor/besca.tl.bcor.postprocess_mnnpy.html">postprocess_mnnpy</a></li>
</ul>
</li>
<li class="toctree-l4"><a class="reference internal" href="#differential-gene-expression">differential gene expression</a><ul>
<li class="toctree-l5"><a class="reference internal" href="dge/besca.tl.dge.perform_dge.html">perform_dge</a></li>
<li class="toctree-l5"><a class="reference internal" href="dge/besca.tl.dge.plot_interactive_volcano.html">plot_interactive_volcano</a></li>
<li class="toctree-l5"><a class="reference internal" href="dge/besca.tl.dge.get_de.html">get_de</a></li>
</ul>
</li>
<li class="toctree-l4"><a class="reference internal" href="#signature-scoring">signature scoring</a><ul>
<li class="toctree-l5"><a class="reference internal" href="sig/besca.tl.sig.combined_signature_score.html">combined_signature_score</a></li>
<li class="toctree-l5"><a class="reference internal" href="sig/besca.tl.sig.compute_signed_score.html">compute_signed_score</a></li>
<li class="toctree-l5"><a class="reference internal" href="sig/besca.tl.sig.filter_siggenes.html">filter_siggenes</a></li>
<li class="toctree-l5"><a class="reference internal" href="sig/besca.tl.sig.read_GMT_sign.html">read_GMT_sign</a></li>
<li class="toctree-l5"><a class="reference internal" href="sig/besca.tl.sig.getset.html">getset</a></li>
<li class="toctree-l5"><a class="reference internal" href="sig/besca.tl.sig.score_mw.html">score_mw</a></li>
<li class="toctree-l5"><a class="reference internal" href="sig/besca.tl.sig.add_anno.html">add_anno</a></li>
<li class="toctree-l5"><a class="reference internal" href="sig/besca.tl.sig.make_anno.html">make_anno</a></li>
<li class="toctree-l5"><a class="reference internal" href="sig/besca.tl.sig.read_annotconfig.html">read_annotconfig</a></li>
<li class="toctree-l5"><a class="reference internal" href="sig/besca.tl.sig.match_cluster.html">match_cluster</a></li>
<li class="toctree-l5"><a class="reference internal" href="sig/besca.tl.sig.obtain_dblabel.html">obtain_dblabel</a></li>
<li class="toctree-l5"><a class="reference internal" href="sig/besca.tl.sig.get_gems.html">get_gems</a></li>
<li class="toctree-l5"><a class="reference internal" href="sig/besca.tl.sig.insert_gems.html">insert_gems</a></li>
<li class="toctree-l5"><a class="reference internal" href="sig/besca.tl.sig.get_similar_geneset.html">get_similar_geneset</a></li>
<li class="toctree-l5"><a class="reference internal" href="sig/besca.tl.sig.export_annotconfig.html">export_annotconfig</a></li>
</ul>
</li>
<li class="toctree-l4"><a class="reference internal" href="#reclustering">reclustering</a><ul>
<li class="toctree-l5"><a class="reference internal" href="reclustering/besca.tl.rc.recluster.html">recluster</a></li>
<li class="toctree-l5"><a class="reference internal" href="reclustering/besca.tl.rc.annotate_new_cellnames.html">annotate_new_cellnames</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#import">Import</a><ul>
<li class="toctree-l3"><a class="reference internal" href="import/besca.Import.read_mtx.html">read_mtx</a></li>
<li class="toctree-l3"><a class="reference internal" href="import/besca.Import.add_cell_labeling.html">add_cell_labeling</a></li>
<li class="toctree-l3"><a class="reference internal" href="import/besca.Import.assert_adata.html">assert_adata</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#export">export</a><ul>
<li class="toctree-l3"><a class="reference internal" href="export/besca.export.X_to_mtx.html">X_to_mtx</a></li>
<li class="toctree-l3"><a class="reference internal" href="export/besca.export.raw_to_mtx.html">raw_to_mtx</a></li>
<li class="toctree-l3"><a class="reference internal" href="export/besca.export.clustering.html">clustering</a></li>
<li class="toctree-l3"><a class="reference internal" href="export/besca.export.labeling.html">labeling</a></li>
<li class="toctree-l3"><a class="reference internal" href="export/besca.export.labeling_info.html">labeling_info</a></li>
<li class="toctree-l3"><a class="reference internal" href="export/besca.export.analysis_metadata.html">analysis_metadata</a></li>
<li class="toctree-l3"><a class="reference internal" href="export/besca.export.generate_gep.html">generate_gep</a></li>
<li class="toctree-l3"><a class="reference internal" href="export/besca.export.ranked_genes.html">ranked_genes</a></li>
<li class="toctree-l3"><a class="reference internal" href="export/besca.export.pseudobulk.html">pseudobulk</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#standardworkflow">standardworkflow</a><ul>
<li class="toctree-l3"><a class="reference internal" href="standardworkflow/besca.st.read_matrix.html">read_matrix</a></li>
<li class="toctree-l3"><a class="reference internal" href="standardworkflow/besca.st.filtering_cells_genes_min.html">filtering_cells_genes_min</a></li>
<li class="toctree-l3"><a class="reference internal" href="standardworkflow/besca.st.filtering_mito_genes_max.html">filtering_mito_genes_max</a></li>
<li class="toctree-l3"><a class="reference internal" href="standardworkflow/besca.st.export_cp10k.html">export_cp10k</a></li>
<li class="toctree-l3"><a class="reference internal" href="standardworkflow/besca.st.export_regressedOut.html">export_regressedOut</a></li>
<li class="toctree-l3"><a class="reference internal" href="standardworkflow/besca.st.export_clustering.html">export_clustering</a></li>
<li class="toctree-l3"><a class="reference internal" href="standardworkflow/besca.st.export_metadata.html">export_metadata</a></li>
<li class="toctree-l3"><a class="reference internal" href="standardworkflow/besca.st.export_rank.html">export_rank</a></li>
<li class="toctree-l3"><a class="reference internal" href="standardworkflow/besca.st.export_celltype.html">export_celltype</a></li>
<li class="toctree-l3"><a class="reference internal" href="standardworkflow/besca.st.additional_labeling.html">additional_labeling</a></li>
<li class="toctree-l3"><a class="reference internal" href="standardworkflow/besca.st.celltype_labeling.html">celltype_labeling</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="auto_examples/index.html">code examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="tutorials.html">tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="besca_standard_pipeline.html">standard pipeline</a></li>
<li class="toctree-l1"><a class="reference internal" href="adding_new_functions.html">adding new functions to besca</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">BESCA</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>besca</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/besca.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast,
.nboutput.nblast {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast + .nbinput {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="besca">
<h1>besca<a class="headerlink" href="#besca" title="Permalink to this headline">Â¶</a></h1>
<div class="section" id="helper-functions">
<span id="id1"></span><h2>helper functions<a class="headerlink" href="#helper-functions" title="Permalink to this headline">Â¶</a></h2>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_raw</span></code></td>
<td></td>
</tr>
<tr class="row-even"><td><code class="xref py py-obj docutils literal notranslate"><span class="pre">subset_adata</span></code></td>
<td></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="preprocessing">
<span id="preprocessing-functions"></span><h2>preprocessing<a class="headerlink" href="#preprocessing" title="Permalink to this headline">Â¶</a></h2>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="preprocessing/besca.pp.filter.html#besca.pp.filter" title="besca.pp.filter"><code class="xref py py-obj docutils literal notranslate"><span class="pre">filter</span></code></a>(adata[,&nbsp;max_genes,&nbsp;min_genes,&nbsp;â€¦])</td>
<td>Filter cell outliers based on counts, numbers of genes expressed, number of cells expressing a gene and mitochondrial gene content.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="preprocessing/besca.pp.filter_gene_list.html#besca.pp.filter_gene_list" title="besca.pp.filter_gene_list"><code class="xref py py-obj docutils literal notranslate"><span class="pre">filter_gene_list</span></code></a>(adata,&nbsp;filepath[,&nbsp;use_raw,&nbsp;â€¦])</td>
<td>Function to remove all genes specified in a gene list read from file.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="preprocessing/besca.pp.frac_pos.html#besca.pp.frac_pos" title="besca.pp.frac_pos"><code class="xref py py-obj docutils literal notranslate"><span class="pre">frac_pos</span></code></a>(adata[,&nbsp;threshold])</td>
<td>Calculate the fraction of cells positive for expression of a gene.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="preprocessing/besca.pp.frac_reads.html#besca.pp.frac_reads" title="besca.pp.frac_reads"><code class="xref py py-obj docutils literal notranslate"><span class="pre">frac_reads</span></code></a>(adata)</td>
<td>Cacluate the fraction of reads being attributed to a specific gene.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="preprocessing/besca.pp.mean_expr.html#besca.pp.mean_expr" title="besca.pp.mean_expr"><code class="xref py py-obj docutils literal notranslate"><span class="pre">mean_expr</span></code></a>(adata)</td>
<td>Calculate the mean expression of a gene.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="preprocessing/besca.pp.top_expressed_genes.html#besca.pp.top_expressed_genes" title="besca.pp.top_expressed_genes"><code class="xref py py-obj docutils literal notranslate"><span class="pre">top_expressed_genes</span></code></a>(adata[,&nbsp;top_n])</td>
<td>Give out the genes most frequently expressed in cells.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="preprocessing/besca.pp.fraction_counts.html#besca.pp.fraction_counts" title="besca.pp.fraction_counts"><code class="xref py py-obj docutils literal notranslate"><span class="pre">fraction_counts</span></code></a>(adata[,&nbsp;species,&nbsp;name,&nbsp;â€¦])</td>
<td>Function to calculate fraction of counts per cell from a gene list.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="preprocessing/besca.pp.top_counts_genes.html#besca.pp.top_counts_genes" title="besca.pp.top_counts_genes"><code class="xref py py-obj docutils literal notranslate"><span class="pre">top_counts_genes</span></code></a>(adata[,&nbsp;top_n])</td>
<td>Give out the genes that contribute the largest fraction to the total UMI counts.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="preprocessing/besca.pp.normalize_geometric.html#besca.pp.normalize_geometric" title="besca.pp.normalize_geometric"><code class="xref py py-obj docutils literal notranslate"><span class="pre">normalize_geometric</span></code></a>(adata)</td>
<td>Perform geometric normalization on CITEseq data.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="plotting">
<span id="plotting-functions"></span><h2>plotting<a class="headerlink" href="#plotting" title="Permalink to this headline">Â¶</a></h2>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="plotting/besca.pl.kp_genes.html#besca.pl.kp_genes" title="besca.pl.kp_genes"><code class="xref py py-obj docutils literal notranslate"><span class="pre">kp_genes</span></code></a>(adata[,&nbsp;threshold,&nbsp;min_genes,&nbsp;ax])</td>
<td>visualize the minimum gene per cell threshold.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="plotting/besca.pl.kp_counts.html#besca.pl.kp_counts" title="besca.pl.kp_counts"><code class="xref py py-obj docutils literal notranslate"><span class="pre">kp_counts</span></code></a>(adata[,&nbsp;min_counts,&nbsp;ax])</td>
<td>visualize the minimum UMI counts per cell threshold.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="plotting/besca.pl.kp_cells.html#besca.pl.kp_cells" title="besca.pl.kp_cells"><code class="xref py py-obj docutils literal notranslate"><span class="pre">kp_cells</span></code></a>(adata[,&nbsp;threshold,&nbsp;min_cells,&nbsp;ax])</td>
<td>visualize the minimum number of cells expressing a gene threshold.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="plotting/besca.pl.max_counts.html#besca.pl.max_counts" title="besca.pl.max_counts"><code class="xref py py-obj docutils literal notranslate"><span class="pre">max_counts</span></code></a>(adata[,&nbsp;max_counts,&nbsp;ax])</td>
<td>visualize maximum UMI counts per cell threshold.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="plotting/besca.pl.max_genes.html#besca.pl.max_genes" title="besca.pl.max_genes"><code class="xref py py-obj docutils literal notranslate"><span class="pre">max_genes</span></code></a>(adata[,&nbsp;max_genes,&nbsp;ax])</td>
<td>visualize maximum number of genes per cell threshold.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="plotting/besca.pl.max_mito.html#besca.pl.max_mito" title="besca.pl.max_mito"><code class="xref py py-obj docutils literal notranslate"><span class="pre">max_mito</span></code></a>(adata[,&nbsp;max_mito,&nbsp;annotation_type,&nbsp;â€¦])</td>
<td>visulize maximum mitochondrial gene percentage threshold.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="plotting/besca.pl.dropouts.html#besca.pl.dropouts" title="besca.pl.dropouts"><code class="xref py py-obj docutils literal notranslate"><span class="pre">dropouts</span></code></a>(adata[,&nbsp;ax,&nbsp;bins])</td>
<td>Plot number of dropouts.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="plotting/besca.pl.detected_genes.html#besca.pl.detected_genes" title="besca.pl.detected_genes"><code class="xref py py-obj docutils literal notranslate"><span class="pre">detected_genes</span></code></a>(adata[,&nbsp;ax,&nbsp;bins])</td>
<td>Plot number of detected genes.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="plotting/besca.pl.library_size.html#besca.pl.library_size" title="besca.pl.library_size"><code class="xref py py-obj docutils literal notranslate"><span class="pre">library_size</span></code></a>(adata[,&nbsp;ax,&nbsp;bins])</td>
<td>Plot library size.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="plotting/besca.pl.librarysize_overview.html#besca.pl.librarysize_overview" title="besca.pl.librarysize_overview"><code class="xref py py-obj docutils literal notranslate"><span class="pre">librarysize_overview</span></code></a>(adata[,&nbsp;bins])</td>
<td>Generates overview figure of libarysize, dropouts and detected genes.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="plotting/besca.pl.transcript_capture_efficiency.html#besca.pl.transcript_capture_efficiency" title="besca.pl.transcript_capture_efficiency"><code class="xref py py-obj docutils literal notranslate"><span class="pre">transcript_capture_efficiency</span></code></a>(adata[,&nbsp;ax])</td>
<td>Plot total gene counts vs detection probability.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="plotting/besca.pl.top_genes_counts.html#besca.pl.top_genes_counts" title="besca.pl.top_genes_counts"><code class="xref py py-obj docutils literal notranslate"><span class="pre">top_genes_counts</span></code></a>(adata[,&nbsp;top_n,&nbsp;ax])</td>
<td>plot top n genes that contribute to fraction of counts per cell</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="plotting/besca.pl.gene_expr_split.html#besca.pl.gene_expr_split" title="besca.pl.gene_expr_split"><code class="xref py py-obj docutils literal notranslate"><span class="pre">gene_expr_split</span></code></a>(adata,&nbsp;genes[,&nbsp;â€¦])</td>
<td>visualize gene expression of two groups as a split violin plot</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="plotting/besca.pl.gene_expr_split_stacked.html#besca.pl.gene_expr_split_stacked" title="besca.pl.gene_expr_split_stacked"><code class="xref py py-obj docutils literal notranslate"><span class="pre">gene_expr_split_stacked</span></code></a>(adata,&nbsp;genes,&nbsp;â€¦[,&nbsp;â€¦])</td>
<td><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">param subset_variable:</th></tr>
<tr class="field-odd field"><td>&#160;</td><td class="field-body">string identifying the column along which the AnnData object should be split into subsets. Can contain as many groups as you wish.</td>
</tr>
</tbody>
</table>
</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="plotting/besca.pl.stacked_split_violin.html#besca.pl.stacked_split_violin" title="besca.pl.stacked_split_violin"><code class="xref py py-obj docutils literal notranslate"><span class="pre">stacked_split_violin</span></code></a>(tidy_data,&nbsp;x_axis,&nbsp;â€¦)</td>
<td>plot stacked split violin plots.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="plotting/besca.pl.celllabel_quant_boxplot.html#besca.pl.celllabel_quant_boxplot" title="besca.pl.celllabel_quant_boxplot"><code class="xref py py-obj docutils literal notranslate"><span class="pre">celllabel_quant_boxplot</span></code></a>(adata,&nbsp;â€¦[,&nbsp;â€¦])</td>
<td>generate a box and whisker plot with overlayed swarm plot of celltype abundances</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="plotting/besca.pl.celllabel_quant_stackedbar.html#besca.pl.celllabel_quant_stackedbar" title="besca.pl.celllabel_quant_stackedbar"><code class="xref py py-obj docutils literal notranslate"><span class="pre">celllabel_quant_stackedbar</span></code></a>(adata,&nbsp;â€¦[,&nbsp;â€¦])</td>
<td>Generate a stacked bar plot of the percentage of labelcounts within each AnnData subset</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="plotting/besca.pl.dot_heatmap.html#besca.pl.dot_heatmap" title="besca.pl.dot_heatmap"><code class="xref py py-obj docutils literal notranslate"><span class="pre">dot_heatmap</span></code></a>(adata,&nbsp;genes[,&nbsp;group_by,&nbsp;â€¦])</td>
<td>Generate a dot plot, filled with heatmap of individuals cells gene expression.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="plotting/besca.pl.dot_heatmap_split.html#besca.pl.dot_heatmap_split" title="besca.pl.dot_heatmap_split"><code class="xref py py-obj docutils literal notranslate"><span class="pre">dot_heatmap_split</span></code></a>(adata,&nbsp;genes,&nbsp;split_by[,&nbsp;â€¦])</td>
<td>Generate a dot plot, filled with heatmap of individuals cells gene expression to compare two conditions.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="plotting/besca.pl.dot_heatmap_split_greyscale.html#besca.pl.dot_heatmap_split_greyscale" title="besca.pl.dot_heatmap_split_greyscale"><code class="xref py py-obj docutils literal notranslate"><span class="pre">dot_heatmap_split_greyscale</span></code></a>(adata,&nbsp;genes,&nbsp;â€¦)</td>
<td>Generate a dot plot, filled with heatmap of individuals cells gene expression to compare two conditions (greyscale).</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="plotting/besca.pl.update_qualitative_palette.html#besca.pl.update_qualitative_palette" title="besca.pl.update_qualitative_palette"><code class="xref py py-obj docutils literal notranslate"><span class="pre">update_qualitative_palette</span></code></a>(adata,&nbsp;palette[,&nbsp;â€¦])</td>
<td>Update adata object such that the umap will adhere to the palette provided.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="plotting/besca.pl.nomenclature_network.html#besca.pl.nomenclature_network" title="besca.pl.nomenclature_network"><code class="xref py py-obj docutils literal notranslate"><span class="pre">nomenclature_network</span></code></a>(config_file[,&nbsp;mywidth,&nbsp;â€¦])</td>
<td>Plot a nomenclature network based on annotation config file.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="tools">
<span id="tools-functions"></span><h2>tools<a class="headerlink" href="#tools" title="Permalink to this headline">Â¶</a></h2>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="tools/besca.tl.count_occurrence.html#besca.tl.count_occurrence" title="besca.tl.count_occurrence"><code class="xref py py-obj docutils literal notranslate"><span class="pre">count_occurrence</span></code></a>(adata[,&nbsp;count_variable,&nbsp;â€¦])</td>
<td>Generate dataframe containing the label counts/percentages of a specific column in adata.obs</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="tools/besca.tl.count_occurrence_subset.html#besca.tl.count_occurrence_subset" title="besca.tl.count_occurrence_subset"><code class="xref py py-obj docutils literal notranslate"><span class="pre">count_occurrence_subset</span></code></a>(adata,&nbsp;subset_variable)</td>
<td>count occurrence of a label in adata.obs after subseting adata object</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="tools/besca.tl.count_occurrence_subset_conditions.html#besca.tl.count_occurrence_subset_conditions" title="besca.tl.count_occurrence_subset_conditions"><code class="xref py py-obj docutils literal notranslate"><span class="pre">count_occurrence_subset_conditions</span></code></a>(adata,&nbsp;â€¦)</td>
<td>count occurrence of a label for each condition in adata.obs after subseting adata object</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="tools/besca.tl.annotate_cells_clustering.html#besca.tl.annotate_cells_clustering" title="besca.tl.annotate_cells_clustering"><code class="xref py py-obj docutils literal notranslate"><span class="pre">annotate_cells_clustering</span></code></a>(adata,&nbsp;â€¦[,&nbsp;â€¦])</td>
<td>Function to add annotation to adata.obs based on clustering This function replaces the original cluster labels located in the column clustering_label with the new values specified in the list new_cluster_lables.</td>
</tr>
</tbody>
</table>
<div class="section" id="toolkits">
<span id="toolkits-functions"></span><h3>toolkits<a class="headerlink" href="#toolkits" title="Permalink to this headline">Â¶</a></h3>
<div class="section" id="batch-correction">
<h4>batch correction<a class="headerlink" href="#batch-correction" title="Permalink to this headline">Â¶</a></h4>
<p>Collection of functions to perform batch correction.</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="besca/besca.tl.bcor.batch_correct.html#besca.tl.bcor.batch_correct" title="besca.tl.bcor.batch_correct"><code class="xref py py-obj docutils literal notranslate"><span class="pre">batch_correct</span></code></a>(adata,&nbsp;batch_to_correct)</td>
<td>function to perform batch correction</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="besca/besca.tl.bcor.postprocess_mnnpy.html#besca.tl.bcor.postprocess_mnnpy" title="besca.tl.bcor.postprocess_mnnpy"><code class="xref py py-obj docutils literal notranslate"><span class="pre">postprocess_mnnpy</span></code></a>(adata,&nbsp;bdata)</td>
<td>postprocessing to generate a newly functional AnnData object</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="differential-gene-expression">
<h4>differential gene expression<a class="headerlink" href="#differential-gene-expression" title="Permalink to this headline">Â¶</a></h4>
<p>Collection of functions to aid in differential gene expression analysis.</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="dge/besca.tl.dge.perform_dge.html#besca.tl.dge.perform_dge" title="besca.tl.dge.perform_dge"><code class="xref py py-obj docutils literal notranslate"><span class="pre">perform_dge</span></code></a>(adata,&nbsp;design_matrix,&nbsp;â€¦[,&nbsp;â€¦])</td>
<td>Perform differential gene expression between two conditions over many adata subsets.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="dge/besca.tl.dge.plot_interactive_volcano.html#besca.tl.dge.plot_interactive_volcano" title="besca.tl.dge.plot_interactive_volcano"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_interactive_volcano</span></code></a>(top_table_path,&nbsp;outdir)</td>
<td>plot an interactive volcano plot based on toptable file.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="dge/besca.tl.dge.get_de.html#besca.tl.dge.get_de" title="besca.tl.dge.get_de"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_de</span></code></a>(adata,&nbsp;mygroup[,&nbsp;demethod,&nbsp;topnr,&nbsp;â€¦])</td>
<td>Get a table of significant DE genes at certain cutoffs Based on an AnnData object and an annotation category (e.g.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="signature-scoring">
<h4>signature scoring<a class="headerlink" href="#signature-scoring" title="Permalink to this headline">Â¶</a></h4>
<p>Collection of functions to aid in signature scoring.</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="sig/besca.tl.sig.combined_signature_score.html#besca.tl.sig.combined_signature_score" title="besca.tl.sig.combined_signature_score"><code class="xref py py-obj docutils literal notranslate"><span class="pre">combined_signature_score</span></code></a>(adata[,&nbsp;GMT_file,&nbsp;â€¦])</td>
<td>Super Wrapper function to compute combined signature score for UP and DN scores.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="sig/besca.tl.sig.compute_signed_score.html#besca.tl.sig.compute_signed_score" title="besca.tl.sig.compute_signed_score"><code class="xref py py-obj docutils literal notranslate"><span class="pre">compute_signed_score</span></code></a>(adata,&nbsp;signature_dict)</td>
<td>Compute signed score combining UP and DN for all signatures in signature_dict This function combines genesets (signatures) scores.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="sig/besca.tl.sig.filter_siggenes.html#besca.tl.sig.filter_siggenes" title="besca.tl.sig.filter_siggenes"><code class="xref py py-obj docutils literal notranslate"><span class="pre">filter_siggenes</span></code></a>(adata,&nbsp;signature_dict)</td>
<td>Filter all signatures in signature_dict to remove genes not present in adata</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="sig/besca.tl.sig.read_GMT_sign.html#besca.tl.sig.read_GMT_sign" title="besca.tl.sig.read_GMT_sign"><code class="xref py py-obj docutils literal notranslate"><span class="pre">read_GMT_sign</span></code></a>(GMT_file[,&nbsp;UP_suffix,&nbsp;â€¦])</td>
<td>Read gmt file to extract signed genesets.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="sig/besca.tl.sig.getset.html#besca.tl.sig.getset" title="besca.tl.sig.getset"><code class="xref py py-obj docutils literal notranslate"><span class="pre">getset</span></code></a>(df,&nbsp;signame_complete,&nbsp;threshold)</td>
<td>Handles missing signatures aux function for make_anno Based on a dataframe of p-values, a signature name and a cutoff check if sign is present :param df: a dataframe of p-values per signature and cell cluster :type df: panda.DataFrame :param signame_complete: signature name :type signame_complete: str :param threshold: cutoff used for cluster attributions :type threshold: numpy.float64</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="sig/besca.tl.sig.score_mw.html#besca.tl.sig.score_mw" title="besca.tl.sig.score_mw"><code class="xref py py-obj docutils literal notranslate"><span class="pre">score_mw</span></code></a>(f,&nbsp;mymarkers)</td>
<td>Score Clusters based on a set of immune signatures to generate a df of pvals Takes as an input a dataframe with fractions per clusters and a dictionary of signatures Performs a Mann-Whitney test per each column and Signature, returns -10logpValues</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="sig/besca.tl.sig.add_anno.html#besca.tl.sig.add_anno" title="besca.tl.sig.add_anno"><code class="xref py py-obj docutils literal notranslate"><span class="pre">add_anno</span></code></a>(adata,&nbsp;cnames,&nbsp;mycol[,&nbsp;clusters])</td>
<td>Adds annotation generated with make_anno to a AnnData object Takes as input the AnnData object to which annotation will be appended and the annotation pd Generates a pd.Series that can be added as a adata.obs column with annotation at a level</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="sig/besca.tl.sig.make_anno.html#besca.tl.sig.make_anno" title="besca.tl.sig.make_anno"><code class="xref py py-obj docutils literal notranslate"><span class="pre">make_anno</span></code></a>(df,&nbsp;sigscores,&nbsp;sigconfig,&nbsp;levsk[,&nbsp;â€¦])</td>
<td>Annotate cell types Based on a dataframe of -log10pvals, a cutoff and a signature set generate cell annotation Hierarchical model of Immune cell annotation.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="sig/besca.tl.sig.read_annotconfig.html#besca.tl.sig.read_annotconfig" title="besca.tl.sig.read_annotconfig"><code class="xref py py-obj docutils literal notranslate"><span class="pre">read_annotconfig</span></code></a>(configfile)</td>
<td>Reads the configuration file for signature-based hierarhical annotation.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="sig/besca.tl.sig.match_cluster.html#besca.tl.sig.match_cluster" title="besca.tl.sig.match_cluster"><code class="xref py py-obj docutils literal notranslate"><span class="pre">match_cluster</span></code></a>(adata,&nbsp;obsquery,&nbsp;obsqueryval)</td>
<td>Matches categories from adata.obs to each other.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="sig/besca.tl.sig.obtain_dblabel.html#besca.tl.sig.obtain_dblabel" title="besca.tl.sig.obtain_dblabel"><code class="xref py py-obj docutils literal notranslate"><span class="pre">obtain_dblabel</span></code></a>(nomenclature_file,&nbsp;cnames)</td>
<td>Matches the cnames obtained by the make_annot function to the db label (standardized label).</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="sig/besca.tl.sig.get_gems.html#besca.tl.sig.get_gems" title="besca.tl.sig.get_gems"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_gems</span></code></a>(setName,&nbsp;BASE_URL[,&nbsp;UP_suffix,&nbsp;â€¦])</td>
<td>Connect to GEMS, dowload related geneset (specified by setName, can be a prefix/suffix) and return them This function combines genesets (signatures) scores (UP and DN) genes.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="sig/besca.tl.sig.insert_gems.html#besca.tl.sig.insert_gems" title="besca.tl.sig.insert_gems"><code class="xref py py-obj docutils literal notranslate"><span class="pre">insert_gems</span></code></a>(BASE_URL,&nbsp;genesets,&nbsp;params[,&nbsp;â€¦])</td>
<td>Insert genesets into the local gems server url_host will depend on GeMs deployement.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="sig/besca.tl.sig.get_similar_geneset.html#besca.tl.sig.get_similar_geneset" title="besca.tl.sig.get_similar_geneset"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_similar_geneset</span></code></a>(request,&nbsp;BASE_URL[,&nbsp;â€¦])</td>
<td>Encapsulating small similary research.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="sig/besca.tl.sig.export_annotconfig.html#besca.tl.sig.export_annotconfig" title="besca.tl.sig.export_annotconfig"><code class="xref py py-obj docutils literal notranslate"><span class="pre">export_annotconfig</span></code></a>(sigconfig,&nbsp;levsk,&nbsp;â€¦[,&nbsp;â€¦])</td>
<td>Export the configuration defined in sigconfig and levsk Order might changed compared to the original sig.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="reclustering">
<h4>reclustering<a class="headerlink" href="#reclustering" title="Permalink to this headline">Â¶</a></h4>
<p>Collection of functions to perform reclustering on selected subclusters.</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="reclustering/besca.tl.rc.recluster.html#besca.tl.rc.recluster" title="besca.tl.rc.recluster"><code class="xref py py-obj docutils literal notranslate"><span class="pre">recluster</span></code></a>(adata,&nbsp;celltype[,&nbsp;celltype_label,&nbsp;â€¦])</td>
<td>Perform subclustering on specific celltype to identify subclusters.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="reclustering/besca.tl.rc.annotate_new_cellnames.html#besca.tl.rc.annotate_new_cellnames" title="besca.tl.rc.annotate_new_cellnames"><code class="xref py py-obj docutils literal notranslate"><span class="pre">annotate_new_cellnames</span></code></a>(adata,&nbsp;â€¦[,&nbsp;â€¦])</td>
<td>annotate new cellnames to each of the subclusters identified by running recluster.</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="section" id="import">
<span id="import-functions"></span><h2>Import<a class="headerlink" href="#import" title="Permalink to this headline">Â¶</a></h2>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="import/besca.Import.read_mtx.html#besca.Import.read_mtx" title="besca.Import.read_mtx"><code class="xref py py-obj docutils literal notranslate"><span class="pre">read_mtx</span></code></a>(filepath[,&nbsp;annotation,&nbsp;use_genes,&nbsp;â€¦])</td>
<td>Read matrix.mtx, genes.tsv, barcodes.tsv to AnnData object.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="import/besca.Import.add_cell_labeling.html#besca.Import.add_cell_labeling" title="besca.Import.add_cell_labeling"><code class="xref py py-obj docutils literal notranslate"><span class="pre">add_cell_labeling</span></code></a>(adata,&nbsp;filepath[,&nbsp;label])</td>
<td>add a labeling written out in the FAIR formating to adata.obs</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="import/besca.Import.assert_adata.html#besca.Import.assert_adata" title="besca.Import.assert_adata"><code class="xref py py-obj docutils literal notranslate"><span class="pre">assert_adata</span></code></a>(adata[,&nbsp;attempFix])</td>
<td>Asserts that an adata object is containing information needed for the besca pipeline to run and export information.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="export">
<span id="export-functions"></span><h2>export<a class="headerlink" href="#export" title="Permalink to this headline">Â¶</a></h2>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="export/besca.export.X_to_mtx.html#besca.export.X_to_mtx" title="besca.export.X_to_mtx"><code class="xref py py-obj docutils literal notranslate"><span class="pre">X_to_mtx</span></code></a>(adata[,&nbsp;outpath,&nbsp;write_metadata,&nbsp;â€¦])</td>
<td>export adata object to mtx format (matrix.mtx, genes.tsv, barcodes.tsv)</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="export/besca.export.raw_to_mtx.html#besca.export.raw_to_mtx" title="besca.export.raw_to_mtx"><code class="xref py py-obj docutils literal notranslate"><span class="pre">raw_to_mtx</span></code></a>(adata[,&nbsp;outpath,&nbsp;write_metadata,&nbsp;â€¦])</td>
<td>export adata.raw to .mtx (matrix.mtx, genes.tsv, barcodes, tsv)</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="export/besca.export.clustering.html#besca.export.clustering" title="besca.export.clustering"><code class="xref py py-obj docutils literal notranslate"><span class="pre">clustering</span></code></a>(adata[,&nbsp;outpath,&nbsp;export_average,&nbsp;â€¦])</td>
<td>export mapping of cells to clusters to .tsv file</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="export/besca.export.labeling.html#besca.export.labeling" title="besca.export.labeling"><code class="xref py py-obj docutils literal notranslate"><span class="pre">labeling</span></code></a>(adata[,&nbsp;outpath,&nbsp;column,&nbsp;label,&nbsp;â€¦])</td>
<td>export mapping of cells to specified label to .tsv file</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="export/besca.export.labeling_info.html#besca.export.labeling_info" title="besca.export.labeling_info"><code class="xref py py-obj docutils literal notranslate"><span class="pre">labeling_info</span></code></a>([outpath,&nbsp;description,&nbsp;â€¦])</td>
<td>write out labeling info for uploading to database</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="export/besca.export.analysis_metadata.html#besca.export.analysis_metadata" title="besca.export.analysis_metadata"><code class="xref py py-obj docutils literal notranslate"><span class="pre">analysis_metadata</span></code></a>(adata[,&nbsp;outpath,&nbsp;â€¦])</td>
<td>export plotting coordinates to analysis_metadata.tsv</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="export/besca.export.generate_gep.html#besca.export.generate_gep" title="besca.export.generate_gep"><code class="xref py py-obj docutils literal notranslate"><span class="pre">generate_gep</span></code></a>(adata[,&nbsp;filename,&nbsp;column,&nbsp;â€¦])</td>
<td>Generate Gene Expression Profile (GEP) from scRNA-seq annotations</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="export/besca.export.ranked_genes.html#besca.export.ranked_genes" title="besca.export.ranked_genes"><code class="xref py py-obj docutils literal notranslate"><span class="pre">ranked_genes</span></code></a>(adata[,&nbsp;type,&nbsp;outpath,&nbsp;â€¦])</td>
<td>export marker genes for each cluster to .gct file</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="export/besca.export.pseudobulk.html#besca.export.pseudobulk" title="besca.export.pseudobulk"><code class="xref py py-obj docutils literal notranslate"><span class="pre">pseudobulk</span></code></a>(adata[,&nbsp;outpath,&nbsp;column,&nbsp;label,&nbsp;â€¦])</td>
<td>export pseudobulk profiles of cells to .gct files</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="standardworkflow">
<span id="standardworkflow-functions"></span><h2>standardworkflow<a class="headerlink" href="#standardworkflow" title="Permalink to this headline">Â¶</a></h2>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="standardworkflow/besca.st.read_matrix.html#besca.st.read_matrix" title="besca.st.read_matrix"><code class="xref py py-obj docutils literal notranslate"><span class="pre">read_matrix</span></code></a>(root_path[,&nbsp;citeseq,&nbsp;â€¦])</td>
<td>Read matrix file as expected for the standard workflow.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="standardworkflow/besca.st.filtering_cells_genes_min.html#besca.st.filtering_cells_genes_min" title="besca.st.filtering_cells_genes_min"><code class="xref py py-obj docutils literal notranslate"><span class="pre">filtering_cells_genes_min</span></code></a>(adata,&nbsp;â€¦)</td>
<td></td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="standardworkflow/besca.st.filtering_mito_genes_max.html#besca.st.filtering_mito_genes_max" title="besca.st.filtering_mito_genes_max"><code class="xref py py-obj docutils literal notranslate"><span class="pre">filtering_mito_genes_max</span></code></a>(adata,&nbsp;â€¦)</td>
<td></td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="standardworkflow/besca.st.export_cp10k.html#besca.st.export_cp10k" title="besca.st.export_cp10k"><code class="xref py py-obj docutils literal notranslate"><span class="pre">export_cp10k</span></code></a>(adata,&nbsp;basepath)</td>
<td>Export raw cp10k to FAIR format for loading into database</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="standardworkflow/besca.st.export_regressedOut.html#besca.st.export_regressedOut" title="besca.st.export_regressedOut"><code class="xref py py-obj docutils literal notranslate"><span class="pre">export_regressedOut</span></code></a>(adata,&nbsp;basepath)</td>
<td>Export regressedOut to FAIR format for loading into database</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="standardworkflow/besca.st.export_clustering.html#besca.st.export_clustering" title="besca.st.export_clustering"><code class="xref py py-obj docutils literal notranslate"><span class="pre">export_clustering</span></code></a>(adata,&nbsp;basepath,&nbsp;method)</td>
<td>Export louvain to cell mapping to FAIR format for loading into database</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="standardworkflow/besca.st.export_metadata.html#besca.st.export_metadata" title="besca.st.export_metadata"><code class="xref py py-obj docutils literal notranslate"><span class="pre">export_metadata</span></code></a>(adata,&nbsp;basepath[,&nbsp;n_pcs,&nbsp;â€¦])</td>
<td>Export metadata in FAIR format for loading into database</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="standardworkflow/besca.st.export_rank.html#besca.st.export_rank" title="besca.st.export_rank"><code class="xref py py-obj docutils literal notranslate"><span class="pre">export_rank</span></code></a>(adata,&nbsp;basepath[,&nbsp;type,&nbsp;â€¦])</td>
<td>Export ranked genes to FAIR format for loading into database</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="standardworkflow/besca.st.export_celltype.html#besca.st.export_celltype" title="besca.st.export_celltype"><code class="xref py py-obj docutils literal notranslate"><span class="pre">export_celltype</span></code></a>(adata,&nbsp;basepath)</td>
<td>Export celltype annotation to cell mapping in FAIR format for loading into database</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="standardworkflow/besca.st.additional_labeling.html#besca.st.additional_labeling" title="besca.st.additional_labeling"><code class="xref py py-obj docutils literal notranslate"><span class="pre">additional_labeling</span></code></a>(adata,&nbsp;labeling_to_use,&nbsp;â€¦)</td>
<td>Standard Workflow function to export an additional labeling besides louvain to FAIR format.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="standardworkflow/besca.st.celltype_labeling.html#besca.st.celltype_labeling" title="besca.st.celltype_labeling"><code class="xref py py-obj docutils literal notranslate"><span class="pre">celltype_labeling</span></code></a>(adata,&nbsp;labeling_author,&nbsp;â€¦)</td>
<td>Standard Workflow function to export an additional labeling besides louvain to FAIR format.</td>
</tr>
</tbody>
</table>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="index.html" class="btn btn-neutral float-left" title="Welcome to bescaâ€™s documentation!" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2020, BEDA

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>